### PR TITLE
rgw: guard memory alloc and fix error handling in RGWBulkDelete

### PR DESCRIPTION
**Pull request by rzarzynski**

### CODE REVIEW

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -4473,18 +4471,25 @@ void RGWBulkDelete::pre_exec()
...
 void RGWBulkDelete::execute()
 {
-  deleter = std::unique_ptr<Deleter>(new Deleter(store, s));
+  try {
+    deleter = std::unique_ptr<Deleter>(\
+          new RGWBulkDelete::Deleter(store, s));
+  } catch (std::bad_alloc) {
+    op_ret = -ENOMEM;
+    return;
+  }
```
Comment (cbodley) [inline src/rgw/rgw_op.cc]: i'm not convinced that catching allocation failures here is a good idea. if we return -ENOMEM at this point, are we actually able to send a reply back to the client without any other allocations? this is an error path that we don't have a way to test. once we're out of memory, there's little we can do to recover gracefully so it's generally best to crash early

Comment (yehudasa) [inline src/rgw/rgw_op.cc]: right, we don't handle in anywhere else

Comment (rzarzynski) [inline src/rgw/rgw_op.cc]: Casey's argumentation convinced me. Although there is is `strdup` where we try to recover through `-ENOMEM`, I think it's much better to exit early. I will rework the PR soon.

**Comment by mattbenjamin** [comment]
@rzarzynski do you have new code for this?  if not, can I close for now?

**Comment by rzarzynski** [comment]
@mattbenjamin: I will upload the rest (like the `op_ret` fix) as a separate PR. Closing this.

**Pull request closed without merging**